neo.UnitTests/SmartContract/UT_ApplicationEngine.cs

```diff
@@ -0,0 +1,185 @@
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Neo.IO;
+using Neo.IO.Caching;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.UnitTests.Ledger;
+using Neo.VM;
+using System;
+
+namespace Neo.UnitTests.SmartContract
+{
+    [TestClass]
+    public class UT_ApplicationEngine
+    {
+        private string message = null;
+        private StackItem item = null;
+        private Store Store;
+
+        [TestInitialize]
+        public void TestSetup()
+        {
+            TestBlockchain.InitializeMockNeoSystem();
+            Store = TestBlockchain.GetStore();
+        }
+
+        [TestMethod]
+        public void TestLog()
+        {
+            ApplicationEngine.Log += Test_Log1;
```

Comment:
Should we remove the `static` to this event?

Reply:
Good idea. It is good to me, no need to care about global effection of the static `Log `in unit test.